### PR TITLE
New roles added and set role endpoint properly registered.

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/config/permissions.neon [hunk 1]
```diff
@@ -168,7 +175,7 @@ permissions:
           - viewGroups
 
     - allow: true
-      role: supervisor
+      role: supervisor-student
```
Comment (Teyras) [inline app/config/permissions.neon]: I don't think a supervisor-student should do anything with instances. Any counterexamples?

File: app/config/permissions.neon [hunk 2]
```diff
@@ -130,7 +137,7 @@ permissions:
 
     - allow: true
       resource: group
-      role: supervisor
+      role: supervisor-student
       actions:
           - supervise
```
Comment (Teyras) [inline app/config/permissions.neon]: This is a really bad privilege name. What can a user with the "supervise" privilege do?

File: app/V1Module/router/RouterFactory.php [hunk 3]
```diff
@@ -331,6 +331,7 @@ private static function createUsersRoutes(string $prefix): RouteList {
     $router[] = new PostRoute("$prefix/<id>", "Users:updateProfile");
     $router[] = new PostRoute("$prefix/<id>/settings", "Users:updateSettings");
     $router[] = new PostRoute("$prefix/<id>/create-local", "Users:createLocalAccount");
+    $router[] = new PostRoute("$prefix/<id>/role", "Users:setRole");
```
Comment (Teyras) [inline app/V1Module/router/RouterFactory.php]: So this was implemented before and we didn't make a route for it?

Comment (krulis-martin) [reply app/V1Module/router/RouterFactory.php]: AFAIK, yes.

Comment (Neloop) [reply app/V1Module/router/RouterFactory.php]: Yes... this is my fault, I forgot the route.

File: app/config/permissions.neon (hunk 2 above)
Comment (krulis-martin) [reply app/config/permissions.neon]: Well, this is not mine and I did not change it. The meaning is that the user may be assigned as a supervisor of a group. I can rename it to "superviseGroup", or if you have better suggestions ...

Comment (Teyras) [reply app/config/permissions.neon]: I know, it's even possible that I wrote it :) How about "becomeSupervisor"?

File: app/V1Module/router/RouterFactory.php (hunk 3 above)
Comment (Teyras) [reply app/V1Module/router/RouterFactory.php]: No hard feelings, just curiosity :)

**Pull request merged**